client/i18n/schema-validation.ts

```diff
@@ -277,6 +291,10 @@ const linksSchemaValidation = languages => {
   });
 };
 
+const getFileJson = (filePath: string) =>
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  require(filePath) as Record<string, object | string>;
```

Comment:
`import` should be used here, and this should be an asynchronous function - all calls would need to be changed.

Reply:
I have modified `getFileJson` to
```ts
const readJsonFile = async (language: string, fileName: string) => {
  const filePath = path.join(
    __dirname,
    `/locales/${language}/${fileName}.json`
  );
  const file = await readFile(filePath, 'utf8');
  const fileJson = JSON.parse(file) as Record<string, unknown>;
  return fileJson;
};
```

And abstract a general method `schemaValidation`

```ts
/**
 * Common Function that checks the json file
 * @param {String} language the language to test
 * @param {String} fileName the fileName of json file to test
 * @param {Object} fileJson the fileJson got by readJsonFile
 * @param {String[]} schemaKeys Array of matching schema's keys
 */
const schemaValidation = (
  language: string,
  fileName: string,
  fileJson: Record<string, unknown>,
  schemaKeys: string[]
) => {/* do the same thing */}
```

Then every schemaValidation will be like

```ts
const linksSchemaValidation = (languages: string[]) => {
  languages.forEach(language => {
    void readJsonFile(language, 'links').then(fileJson => {
      schemaValidation(language, 'links', fileJson, linksSchemaKeys);
    });
  });
};
```